lib/ash/type/type.ex

```diff
@@ -1285,12 +1285,23 @@ defmodule Ash.Type do
     end
   end
 
+  @doc false
+  defp remove_empty_items(term, constraints) do
+    empty_values = Keyword.get(constraints, :empty_values, [])
```

Comment:
lets create one list up front of empty values to be removed. If the list is empty, we don't need to do `Enum.filter` we can just return the `term` unchanged.

Reply:
Hm...well...doesn't this mean we're doubling up the remove empty items logic? One place to remove it, and once place to remove it and check length? wouldn't this issue just be an "order of operations" issue in that case? remove empty items before checking length?